src/RA_Leaderboard.cpp

```diff
@@ -97,40 +89,16 @@ void RA_Leaderboard::Submit(unsigned int nScore)
 
 void RA_Leaderboard::SubmitRankInfo(unsigned int nRank, const std::string& sUsername, int nScore, time_t nAchieved)
 {
-    Entry newEntry;
-    newEntry.m_nRank = nRank;
-    newEntry.m_sUsername = sUsername;
-    newEntry.m_nScore = nScore;
-    newEntry.m_TimeAchieved = nAchieved;
-
-    std::vector<Entry>::iterator iter = m_RankInfo.begin();
-    while (iter != m_RankInfo.end())
-    {
-        if ((*iter).m_nRank == nRank)
-        {
-            (*iter) = newEntry;
-            return;
-        }
-        iter++;
-    }
+    if (EntryExists(nRank))
+        return;
```

Comment:
This assumes the list is always sorted, which implies that it needs to be resorted after the emplace_back, or the item needs to be inserted in the correct place.

Reply:
I probably shouldn't have moved it to another function, but it finds it via binary search

Reply:
Yes, I see that, but BinarySearch only works if the list is sorted. This method only appends to the list and Sort is called later. If the items aren't added in order, the BinarySearch may fail.